enterprise/server/cmd/ci_runner/main.go

```diff
@@ -587,26 +630,52 @@ func setupGitRepo(ctx context.Context) error {
 		return err
 	}
 	remote, err := repo.CreateRemote(&gitcfg.RemoteConfig{
-		Name: "origin",
+		Name: remoteName,
 		URLs: []string{authURL},
 	})
 	if err != nil {
 		return err
 	}
-
-	fetchOpts := &git.FetchOptions{
-		RefSpecs: []gitcfg.RefSpec{gitcfg.RefSpec(fmt.Sprintf("+refs/heads/%s:refs/remotes/origin/%s", *branch, *branch))},
-	}
-	if err := remote.FetchContext(ctx, fetchOpts); err != nil {
+	if err := fetchOrigin(ctx, remote); err != nil {
 		return err
 	}
 	tree, err := repo.Worktree()
 	if err != nil {
-		return err
+		return status.WrapErrorf(err, "get tree")
 	}
 	if err := tree.Checkout(&git.CheckoutOptions{Hash: gitplumbing.NewHash(*commitSHA)}); err != nil {
+		return status.WrapErrorf(err, "checkout %s", *commitSHA)
+	}
+	return nil
+}
+
+func syncExistingRepo(ctx context.Context, path string) error {
+	log.Printf("Attempting to set up existing git repo")
```

Comment:
rm

Reply:
(maybe add a log at the end, on success, if you want to keep this?)

Reply:
(Removed -- this was mostly for debugging purposes)